core/services/validators.go

```diff
@@ -38,6 +39,16 @@ func ValidateJob(j models.JobSpec, store *store.Store) error {
 	return fe.CoerceEmptyToNil()
 }
 
+// ValidateBridgeTypeNotExist checks that a bridge has not already been created
+func ValidateBridgeTypeNotExist(bt *models.BridgeTypeRequest, store *store.Store) error {
```

Comment:
Is this backed by a uniqueness constraint?

Reply:
It is, but it doesn't return a very user friendly error message.

Reply:
I also feel like this could be moved to Create(..) by defining a new error for signalling a record already exists.  Maybe we could discuss this at some point

Reply:
That's fine, but that error could still be returned in this PR, so I think you'd still want to match against the DB error and if it returns a uniqueness constraint error transform it into this friendlier error you've added.

Reply:
Also, are you planning on moving the validation into Create? I think that'd be a fine change and work nicely with the change I've suggested in my previous comment.

Reply:
I would have down this, but [Gorm](https://godoc.org/github.com/jinzhu/gorm#pkg-variables) does define these 'uniqueness constraint' errors to match against.  So from what I'm aware, this is not possible.

Reply:
You can match against the string, it's ugly but it works and we've done it in two other places, e.g: `noSuchTableRegex = regexp.MustCompile(`^(no such table|pq: relation ".*?" does not exist)`)`.

Reply:
I've remember coming across somewhere this is a faux pas in Golang as you can't rely on the error's string to remain the same, its not a codified part of the package.  Otherwise the implementers would have exported the error or something similar.  I would really hesitate to do this.

Reply:
If it doesn't match the worst you get is an error message that is not as friendly and you save on extra error calls and eliminate a race condition.

Reply:
The problem from my perspective is its propagating very bad error handling practices to improve an error message which will affect a tiny group of users, if at all.  But going ahead with this for now.

There may be another approach where we can check the error codes of the underlying sqlite or postgres errors, but it would take some work and best handled on its own.